### ISSUE TITLE
[a11y]: The Truncated text component seems to lack a keyboard affordance

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#21189]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://ibm-products.carbondesignsystem.com/?path=/docs/utilities-truncatedtext--overview#example-usage

### Automated testing tool and ruleset

n/a

### Description

Although a mouse user can hover over a truncated text paragraph to have the truncated text displayed in a tooltip, there seems to be no way of triggering by keyboard.

### WCAG 2.1 Violation

2.1.1 Keyboard

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/docs/utilities-truncatedtext--overview#example-usage

### Steps to reproduce

1. Go to https://ibm-products.carbondesignsystem.com/?path=/docs/utilities-truncatedtext--overview#example-usage
2. On the Overview page, attempt to reach the paragraph that ends in an ellipsis via keyboard

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
